src/selfmain.c

```diff
@@ -79,31 +90,43 @@ void setBootProt(int v) {
         return;
     }
 
-    fuses[0] = (fuses[0] & ~NVMCTRL_FUSES_BOOTPROT_Msk) | (v << NVMCTRL_FUSES_BOOTPROT_Pos);
+    newfuses[0] = (newfuses[0] & ~NVMCTRL_FUSES_BOOTPROT_Msk) | (v << NVMCTRL_FUSES_BOOTPROT_Pos);
 
-    #ifdef SAMD21
-    NVMCTRL->CTRLB.reg = NVMCTRL->CTRLB.reg | NVMCTRL_CTRLB_CACHEDIS | NVMCTRL_CTRLB_MANW;
+    const bool format = false;
+    for (int i = 0; i < sizeof(newfuses); ++i)
+        format |= ((newfuses[i] ^ fuses[i]) & newfuses[i]);
```

Comment:
This does not compile because `format` is `const`. Travis didn't find this because the Makefile is not propagating status correctly.

Please do test the compile for all boards after fixing. Thanks.

Reply:
`make BOARD=metro_m4` will make it obvious.

Reply:
If you add the `-1` to these lines in `Makefile` the Travis check will fail properly. Thanks!
```
all-boards:
	for f in `cd boards; ls` ; do "$(MAKE)" BOARD=$$f drop-board || break -1; done
```

Reply:
Cost has been fixed.. Not clear all other notes...

Reply:
makefile fixed